components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -1150,4 +1224,73 @@ private void provisionSCIMOperation(int provisioningMethod, SCIMObject provision
             throw new CharonException("Error in initializing provisioning handler", e);
         }
     }
+
+    private void provision(ProvisioningOperation provisioningMethod, SCIMObject provisioningObject) throws CharonException {
+
+        String domainName = UserCoreUtil.getDomainName(carbonUM.getRealmConfiguration());
+        try {
+
+            Map<org.wso2.carbon.identity.application.common.model.ClaimMapping, List<String>> outboundAttributes = new HashMap<org.wso2.carbon.identity.application.common.model.ClaimMapping, List<String>>();
+
+            if (provisioningObject.getAttribute("password") != null) {
+                outboundAttributes.put(org.wso2.carbon.identity.application.common.model.ClaimMapping.build(
+                                IdentityProvisioningConstants.PASSWORD_CLAIM_URI, null, null, false),
+                        Arrays.asList(new String[]{((SimpleAttribute) provisioningObject.getAttribute("password"))
+                                .getStringValue()}));
+            }
+
+            if (provisioningObject.getAttribute(USER_NAME_STRING) != null) {
```

Comment:
Here the code is checking for provisioningObject.getAttribute(USER_NAME_STRING) !=null.But before that isn't it we need to check provisioningObject itself also not null ?